omnigibson/object_states/toggle.py

```diff
@@ -19,19 +19,58 @@
 m.CAN_TOGGLE_STEPS = 5
 
 
-class ToggledOn(AbsoluteObjectState, BooleanStateMixin, LinkBasedStateMixin, UpdateStateMixin):
+class ToggledOn(AbsoluteObjectState, BooleanStateMixin, LinkBasedStateMixin, UpdateStateMixin, GlobalUpdateStateMixin):
+
+    # Set of prim paths defining robot finger links belonging to any manipulation robots
+    _robot_finger_paths = None
+
+    # Set of objects that are overlapping any manipulation robots
+    _finger_overlap_objs = None
+
     def __init__(self, obj, scale=None):
         self.scale = scale
         self.value = False
         self.robot_can_toggle_steps = 0
         self.visual_marker = None
-        self._check_overlap = None
-        self._robot_link_paths = None
 
         # We also generate the function for checking overlaps at runtime
+        self._check_overlap = None
 
         super().__init__(obj)
 
+    @classmethod
+    def global_update(cls):
+        # Avoid circular imports
+        from omnigibson.robots.manipulation_robot import ManipulationRobot
+
+        # Clear finger overlap objects since it will be refreshed now
+        cls._finger_overlap_objs = set()
+
+        # detect marker and hand interaction
+        robot_finger_links = set(link
+                                 for robot in og.sim.scene.robots if isinstance(robot, ManipulationRobot)
+                                 for finger_links in robot.finger_links.values()
+                                 for link in finger_links)
+        cls._robot_finger_paths = set(link.prim_path for link in robot_finger_links)
```

Comment:
this is not detecting marker and hand interaction, right? It's detecting overlap between hand and any other objects. Maybe add a few more comments to this function to be very explicit? 

Also, I assumed you tested this, but when a robot finger touches an object, that object is guaranteed to show up in the overlap callback? Does it require some sort of penetration?